core/src/main/java/hudson/cli/CLICommand.java

```diff
@@ -293,13 +292,7 @@ public int main(List<String> args, Locale locale, InputStream stdin, PrintStream
             stderr.println("ERROR: Bad Credentials. Search the server log for " + id + " for more details.");
             return 7;
         } catch (Throwable e) {
-            final String errorMsg = String.format("Unexpected exception occurred while performing %s command.",
-                    getName());
-            stderr.println();
-            stderr.println("ERROR: " + errorMsg);
-            LOGGER.log(Level.WARNING, errorMsg, e);
-            Functions.printStackTrace(e, stderr);
-            return 1;
+            return CLIReportUnexpectedException.report(getName(), LOGGER, stderr, e);
```

Comment:
It could be useful to let the `return 1` aside, to ease the return code finding. It's particularly difficult to know where the numbers come from at the first place. Putting it in a deeper method (in different class) will not help there.

Reply:
#4076